c/tskit/genotypes.c

```diff
@@ -416,6 +413,55 @@ tsk_variant_update_genotypes_traversal(
     return tsk_variant_traverse(self, node, derived, tsk_variant_visit);
 }
 
+static void
+tsk_variant_update_missing_cache(tsk_variant_t *self)
+{
+    const tsk_id_t *restrict left_child = self->tree.left_child;
+    const tsk_id_t *restrict right_sib = self->tree.right_sib;
+    const tsk_id_t *restrict sample_index_map = self->sample_index_map;
+    const tsk_id_t N = self->tree.virtual_root;
+    const tsk_flags_t *restrict flags = self->tree_sequence->tables->nodes.flags;
+    bool *restrict present = self->sample_is_present;
+    tsk_id_t *restrict stack = self->traversal_stack;
+    tsk_id_t root, u, v, sample_index;
+    tsk_size_t j;
+    int stack_top;
+
+    tsk_bug_assert(self->alt_samples != NULL);
+    tsk_bug_assert(self->tree.index != TSK_NULL);
+
+    tsk_memset(present, 0, self->num_samples * sizeof(*present));
+
+    stack_top = -1;
```

Comment:
This is probably a bit dumb, but why do we need to do the traversal? Can't we just do the ``num_samples`` loop below, check each sample to see if it's isolated?